src/TLuaInterpreter.cpp

```diff
@@ -14594,6 +14594,26 @@ int TLuaInterpreter::getOS(lua_State* L)
     return 1;
 }
 
+int TLuaInterpreter::getClipboardText(lua_State* L)
+{
+    QClipboard* clipboard = QApplication::clipboard();
+    lua_pushstring(L, clipboard->text().toUtf8().constData());
+    return 1;
+}
+
+int TLuaInterpreter::setClipboardText(lua_State* L)
+{
+    if (lua_isstring(L, 1)) {
+        QClipboard* clipboard = QApplication::clipboard();
+        clipboard->setText(QString::fromUtf8(lua_tostring(L, 1)));
+        lua_pushboolean(L, true);
+        return 1;
+    } else {
```

Comment:
This setup seems fine to me.

Reply:
It is functionally fine - but could be better written as:
```cpp
int TLuaInterpreter::setClipboardText(lua_State* L)
{
    if (lua_isstring(L, 1)) {
        QClipboard* clipboard = QApplication::clipboard();
        clipboard->setText(QString::fromUtf8(lua_tostring(L, 1)));
        lua_pushboolean(L, true);
        return 1;
    }
    lua_pushfstring(L, "setClipboard: bad argument #%d type (text as string expected, got %s!)", 1, luaL_typename(L, 1));
    return lua_error(L);
}
```

and given our past practice of *failing early* it would more closely match other functions as:
```cpp
int TLuaInterpreter::setClipboardText(lua_State* L)
{
    if (lua_isstring(L, 1)) {
        lua_pushfstring(L, "setClipboard: bad argument #%d type (text as string expected, got %s!)", 1, luaL_typename(L, 1));
        return lua_error(L);
    }
    QClipboard* clipboard = QApplication::clipboard();
    clipboard->setText(QString::fromUtf8(lua_tostring(L, 1)));
    lua_pushboolean(L, true);
    return 1;
}
```

Reply:
That is true :+1:

Reply:
Updated!